stable_baselines3/common/preprocessing.py

```diff
@@ -62,11 +62,21 @@ def preprocess_obs(obs: th.Tensor, observation_space: spaces.Space,
         if is_image_space(observation_space) and normalize_images:
             return obs.float() / 255.0
         return obs.float()
+
     elif isinstance(observation_space, spaces.Discrete):
         # One hot encoding and convert to float to avoid errors
         return F.one_hot(obs.long(), num_classes=observation_space.n).float()
+
+    elif isinstance(observation_space, spaces.MultiDiscrete):
+        # Tensor concatination of one hot encodings of each Categorical sub-space
+        return th.cat([F.one_hot(o.long(), num_classes=int(observation_space.nvec[i])).float()
+                       for i, o in enumerate(th.split(obs.to(th.int64), 1, dim=1))],
+                      dim=-1).view(obs.shape[0], sum(observation_space.nvec))
```

Comment:
```suggestion
        return th.cat([F.one_hot(obs_, num_classes=int(observation_space.nvec[idx])).float()
                       for idx, obs_ in enumerate(th.split(obs.long(), 1, dim=1))],
                      dim=-1).view(obs.shape[0], sum(observation_space.nvec))
```

Again, I removed one letter variables and used `.long()` because `to(th.int64)` won't work on GPU, no?